library/Requests/Transport/fsockopen.php

```diff
@@ -56,6 +56,25 @@ class Requests_Transport_fsockopen implements Requests_Transport {
 	 * @return string Raw HTTP result
 	 */
 	public function request($url, $headers = array(), $data = array(), $options = array()) {
+		if (is_array($data) === false && is_string($data) === false) {
+			if ($data === null) {
+				$data = '';
+			} elseif (is_int($data) || is_float($data)) {
+				$data = (string) $data;
+			} else {
+				throw new Requests_Exception_InvalidArgument(
+					sprintf(
+						'%s: Argument #%d (%s) must be of type %s, %s given',
+						__METHOD__,
+						3,
+						'$data',
+						'array|string',
+						gettype($data)
+					)
+				);
+			}
+		}
```

Comment:
As the entire check is duplicated, I suggest we extract this into a protected method of the base transport class:
```suggestion
		$data = $this->validate_data_tyoe($data);
```

Reply:
We don't have a base class, only an interface. So we'll skip this refactor for now.